src/transformers/pipelines.py

```diff
@@ -454,14 +454,17 @@ def ensure_tensor_on_device(self, **inputs):
         """
         return {name: tensor.to(self.device) for name, tensor in inputs.items()}
 
-    def _parse_and_tokenize(self, *args, pad_to_max_length=True, **kwargs):
+    def _parse_and_tokenize(self, *args, pad_to_max_length=True, add_special_tokens=True, **kwargs):
```

Comment:
Currently, all inputs are tokenized with the option `add_special_tokens=True`. This should not be the case for text generation though. `XLNet` for example appends the tokens <seep> and <cls> to the input which leads to bad generation quality. See:
```python
from transformers import XLNetTokenizer
tok = XLNetTokenizer.from_pretrained("xlnet-base-cased") 
tok.decode(tok.encode("Hello"))  # gives "Hello<sep><cls>"
```